operators/wireguard/internal/controllers/cluster-connection/controller.go

```diff
@@ -267,6 +277,38 @@ func (r *Reconciler) reconGateway(req *rApi.Request[*wgv1.ClusterConnection]) st
 	ctx, obj := req.Context(), req.Object
 	check := rApi.NewRunningCheck(GWReady, req)
 
+	corefile := ""
+	kubeDns, err := rApi.Get(ctx, r.Client, fn.NN("kube-system", "kube-dns"), &corev1.Service{})
+	if err != nil {
+		return check.Failed(err)
+	}
+
+	corefile += fmt.Sprintf("\n\trewrite name regex (^[a-zA-Z0-9-_]+)[.]([a-zA-Z0-9-_]+)[.]svc[.]cluster%d[.]local {1}.{2}.svc.%s answer auto",
+		obj.Spec.Id,
+		r.Env.ClusterInternalDns,
+	)
+
+	corefile += fmt.Sprintf("\n\trewrite name regex (^[a-zA-Z0-9-_]+)[.]([a-zA-Z0-9-_]+)[.]cluster%d[.]local {1}.{2}.svc.%s answer auto",
+		obj.Spec.Id,
+		r.Env.ClusterInternalDns,
+	)
+
+	corefile = fmt.Sprintf(`
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Consider extracting CoreDNS config generation to a separate method.

The logic for generating the CoreDNS configuration is quite extensive and could benefit from being encapsulated in its own method for better readability and maintainability.